pkg/typesystem/typesystem.go

```diff
@@ -331,6 +331,32 @@ func (t *TypeSystem) DirectlyRelatedUsersets(objectType, relation string) ([]*op
 	return usersetRelationReferences, nil
 }
 
+// resolvesTypeRelationToDirectlyAssignable returns whether the assignable types ONLY if
+// the specified type and relation is exclusively assignable. Otherwise, it will return nil as slice.
+func (t *TypeSystem) resolvesTypeRelationToDirectlyAssignable(objectType, relationName string) ([]string, bool, error) {
+	relation, err := t.GetRelation(objectType, relationName)
+	if err != nil {
+		return nil, false, err
+	}
+	_, ok := relation.GetRewrite().GetUserset().(*openfgav1.Userset_This)
+	if !ok {
+		return nil, false, err
+	}
+
+	directlyRelatedTypes := relation.GetTypeInfo().GetDirectlyRelatedUserTypes()
+	assignableTypes := make([]string, 0, len(directlyRelatedTypes))
+	// need to check whether these are simple types as well
+	for _, ref := range directlyRelatedTypes {
+		if ref.GetRelationOrWildcard() != nil || ref.GetCondition() != "" {
+			// For now, we don't allow if these types are another userset, publicly assignable or has condition
+			// because local check with these relations cannot be evaluated via simple datastore query.
+			return nil, false, nil
+		}
+		assignableTypes = append(assignableTypes, ref.GetType())
+	}
+	return assignableTypes, true, nil
```

Comment:
Why the slice? To cache later?

Reply:
Got it, you need it for TTU. Lets mark this part to be memoized in subsequent calls given that it will happen _a lot_.